### PR TITLE
refactor blockchain/v0 tests to use p2ptest

### PR DESCRIPTION
**Pull request by tychoish**
this is mostly like the last couple of refactors, with the additional
changes to support more clearly adding nodes to existing p2ptest
networks.

### CODE REVIEW

File: p2p/p2ptest/network.go [hunk 1]
```diff
@@ -197,8 +197,8 @@ type Node struct {
 	Transport   *p2p.MemoryTransport
 }
 
-// MakeNode creates a new Node.
-func MakeNode(t *testing.T, network *Network) *Node {
+// AddNode creates a new Node and adds it to the network.
```
Comment (cmwaters) [inline p2p/p2ptest/network.go:200]: but is it adding it to the network?

**Review by cmwaters — APPROVED** [review]
utAck 👍

Side question: do we want to disable `nolintlint`?

File: p2p/p2ptest/network.go (hunk 1 above)
Comment (tychoish) [reply p2p/p2ptest/network.go]: yeah, good catch, I've updated the docstring and renamed the function.

File: p2p/p2ptest/network.go [hunk 2]
```diff
@@ -197,8 +197,10 @@ type Node struct {
 	Transport   *p2p.MemoryTransport
 }
 
-// MakeNode creates a new Node.
-func MakeNode(t *testing.T, network *Network) *Node {
+// MakeNode creates a new Node configured for the network with a
+// running peer manager, but does not add it to the existing
+// network. Callers are responsible for updating peering relationships
```
Comment (alexanderbez) [inline p2p/p2ptest/network.go:202]:
```suggestion
// network. Callers are responsible for updating peering relationships.
```

**Review by alexanderbez — APPROVED** [review]
🎉

**Pull request merged**